cime_config/buildnml

```diff
@@ -122,383 +183,157 @@ def _create_drv_namelists(case, infile, confdir, nmlgen, files):
-            "user_xml_dir %s does not exist " %user_xml_dir)
+    # Set namelist definition file
+    definition_file = [os.path.join(namelist_xml_dir, "namelist_definition_drv.xml")]
 
-    files = Files(comp_interface="nuopc")
-
-    # TODO: to get the right attributes of COMP_ROOT_DIR_CPL in evaluating definition_file - need
-    # to do the following first - this needs to be changed so that the following two lines are not needed!
-    comp_root_dir_cpl = files.get_value( "COMP_ROOT_DIR_CPL",{"component":"drv-nuopc"}, resolved=False)
-    files.set_value("COMP_ROOT_DIR_CPL", comp_root_dir_cpl)
-
-    definition_file = [files.get_value("NAMELIST_DEFINITION_FILE", {"component": "drv-nuopc"})]
-    fd_dir = os.path.dirname(definition_file[0])
-    user_definition = os.path.join(user_xml_dir, "namelist_definition_drv.xml")
-    if os.path.isfile(user_definition):
-        definition_file = [user_definition]
-
-    # create the namelist generator object - independent of instance
+    # Create the namelist generator object - independent of instance
     nmlgen = NamelistGenerator(case, definition_file)
 
-    # create cplconf/namelist
+    # Create cplconf/namelist
     infile_text = ""
 
-    # determine infile list for nmlgen
+    # Determine infile list for nmlgen
     user_nl_file = os.path.join(caseroot, "user_nl_cpl")
     namelist_infile = os.path.join(confdir, "namelist_infile")
     create_namelist_infile(case, user_nl_file, namelist_infile, infile_text)
     infile = [namelist_infile]
 
-    # create the files nuopc.runconfig, nuopc.runseq, drv_in and drv_flds_in
-    _create_drv_namelists(case, infile, confdir, nmlgen, files)
+    # Create the files nuopc.runconfig, nuopc.runseq, drv_in and drv_flds_in
+    _create_drv_namelists(case, infile, confdir, nmlgen)
 
-    # create the files comp_modelio.nml where comp = [atm, lnd...]
-    _create_component_modelio_namelists(confdir, case, files)
-
-    # set rundir
+    # Set rundir
     rundir = case.get_value("RUNDIR")
 
-    # copy nuopc.runconfig to rundir
+    # Copy nuopc.runconfig to rundir
+    shutil.copy(os.path.join(confdir,"nems.configure"), rundir)
+
+    # Prepare input
+    prep_input(case, casedocsdir, nmlgen)
+
+    # Copy fd.yaml to rundir
+    cimeroot = case.get_value("CIMEROOT")
+    fd_dir = os.path.join(cimeroot,"src","drivers","nuopc","mediator")
+    if not os.path.isdir(fd_dir):
+        srcroot = case.get_value("SRCROOT")
+        fd_dir = os.path.join(srcroot,"src","model","CMEPS","mediator")
+
+    coupling_mode = case.get_value('COUPLING_MODE')
+    if coupling_mode == 'hafs':
+        filename = os.path.join(fd_dir,"fd_hafs.yaml")
+        shutil.copy(filename, os.path.join(rundir, "fd.yaml"))
+    elif 'nems' in coupling_mode:
+        filename = os.path.join(fd_dir,"fd_nems.yaml")
+        shutil.copy(filename, os.path.join(rundir, "fd_nems.yaml"))
+    else:
+        expect(False, "coupling mode currently only supports cesm, hafs and nems")
```

Comment:
print the coupling mode:    expect(False, "coupling mode {} not in (cesm, hafs and nems)".format(coupling_mode))

Reply:
done